ac.soton.emf.translator/src/ac/soton/emf/translator/TranslatorFactory.java

```diff
@@ -34,17 +35,33 @@
 import ac.soton.emf.translator.impl.TranslateCommand;
 import ac.soton.emf.translator.impl.Translator;
 import ac.soton.emf.translator.impl.TranslatorConfig;
+import ac.soton.emf.translator.impl.UnTranslateCommand;
 
 
 public class TranslatorFactory {
 
 	// The shared instance
 	private static TranslatorFactory factory = null;
 	
-	
 	//cached store of translator configurations that have been loaded from extension points
 	private Map<String,TranslatorConfig> translatorConfigRegistry = new HashMap<String, TranslatorConfig  >();
 	
+	//map of the translators available for each command
+	private Map<String,List<String> > translatorsForCommand = new HashMap<String, List<String>>();
```

Comment:
Can translatorsForCommand be *Map<String,Collection<String>>* instead?

Reply:
Yes, Collection is better and use HashSet for creating a new one.
I also added a check for conflicting translators (same commandID and rootSourceClass) and ignore the second and log an error